Cannot build (cross) for aarch64-linux-android

```
$ cargo build --target=aarch64-linux-android --release
   Compiling users v0.5.3
   Compiling libgit2-sys v0.6.13
   Compiling num v0.1.40
   Compiling regex v0.1.80
error[E0412]: cannot find type `c_char` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:75:23
   |
75 |     gr_name:   *const c_char,         // group name
   |                       ^^^^^^ did you mean `char`?
   |
help: possible candidates are found in other modules, you can import them into scope
   | use libc::c_char;
   | use std::os::raw::c_char;

error[E0412]: cannot find type `c_char` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:76:23
   |
76 |     gr_passwd: *const c_char,         // password
   |                       ^^^^^^ did you mean `char`?
   |
help: possible candidates are found in other modules, you can import them into scope
   | use libc::c_char;
   | use std::os::raw::c_char;

error[E0412]: cannot find type `c_char` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:78:30
   |
78 |     gr_mem:    *const *const c_char,  // names of users in the group
   |                              ^^^^^^ did you mean `char`?
   |
help: possible candidates are found in other modules, you can import them into scope
   | use libc::c_char;
   | use std::os::raw::c_char;

error[E0412]: cannot find type `c_passwd` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:82:39
   |
82 |     fn getpwuid(uid: uid_t) -> *const c_passwd;
   |                                       ^^^^^^^^ not found in this scope

error[E0412]: cannot find type `c_char` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:83:35
   |
83 |     fn getpwnam(user_name: *const c_char) -> *const c_passwd;
   |                                   ^^^^^^ did you mean `char`?
   |
help: possible candidates are found in other modules, you can import them into scope
   | use libc::c_char;
   | use std::os::raw::c_char;

error[E0412]: cannot find type `c_passwd` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:83:53
   |
83 |     fn getpwnam(user_name: *const c_char) -> *const c_passwd;
   |                                                     ^^^^^^^^ not found in this scope

error[E0412]: cannot find type `c_char` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:86:36
   |
86 |     fn getgrnam(group_name: *const c_char) -> *const c_group;
   |                                    ^^^^^^ did you mean `char`?
   |
help: possible candidates are found in other modules, you can import them into scope
   | use libc::c_char;
   | use std::os::raw::c_char;

error[E0412]: cannot find type `c_passwd` in this scope
  --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:95:29
   |
95 |     fn getpwent() -> *const c_passwd;
   |                             ^^^^^^^^ not found in this scope

error[E0412]: cannot find type `UserExtras` in module `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:105:17
    |
105 |     extras: os::UserExtras,
    |                 ^^^^^^^^^^ not found in `os`

error[E0433]: failed to resolve. Could not find `UserExtras` in `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:124:21
    |
124 |             extras: os::UserExtras::default(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^ Could not find `UserExtras` in `os`

error[E0412]: cannot find type `GroupExtras` in module `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:165:17
    |
165 |     extras: os::GroupExtras,
    |                 ^^^^^^^^^^^ not found in `os`

error[E0433]: failed to resolve. Could not find `GroupExtras` in `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:183:21
    |
183 |             extras: os::GroupExtras::default(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `GroupExtras` in `os`

error[E0412]: cannot find type `c_char` in this scope
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:220:34
    |
220 | unsafe fn from_raw_buf(p: *const c_char) -> String {
    |                                  ^^^^^^ did you mean `char`?
    |
help: possible candidates are found in other modules, you can import them into scope
    | use libc::c_char;
    | use std::os::raw::c_char;

error[E0412]: cannot find type `c_passwd` in this scope
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:239:42
    |
239 | unsafe fn passwd_to_user(pointer: *const c_passwd) -> Option<User> {
    |                                          ^^^^^^^^ not found in this scope

error[E0433]: failed to resolve. Could not find `UserExtras` in `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:247:28
    |
247 |             extras:        os::UserExtras::from_passwd(passwd),
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `UserExtras` in `os`

error[E0433]: failed to resolve. Could not find `GroupExtras` in `os`
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:262:24
    |
262 |             extras:    os::GroupExtras::from_struct(group),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `GroupExtras` in `os`

error[E0412]: cannot find type `c_char` in this scope
   --> /home/nathan/.cargo/registry/src/github.com-1ecc6299db9ec823/users-0.5.3/src/base.rs:277:41
    |
277 | unsafe fn members(groups: *const *const c_char) -> Vec<String> {
    |                                         ^^^^^^ did you mean `char`?
    |
help: possible candidates are found in other modules, you can import them into scope
    | use libc::c_char;
    | use std::os::raw::c_char;
```

--- a/src/base.rs
+++ b/src/base.rs
@@ -48,13 +48,17 @@ use libc::{
     geteuid,
     getgid,
     getegid,
-    setpwent,
-    getpwent,
-    endpwent,
     getgroups,
     getgrouplist,
     c_int,
 };
+#[cfg(not(target_os = "android"))]
+use libc::{
+    setpwent,
+    getpwent,
+    endpwent
+};
+
 use libc::passwd as c_passwd;
 use libc::group as c_group;
 
@@ -415,12 +419,14 @@ struct AllUsers;
 /// block, then make sure to not make a new instance of it until
 /// iteration is over.
 pub unsafe fn all_users() -> impl Iterator<Item=User> {
+    #[cfg(not(target_os = "android"))]
     setpwent();
     AllUsers
 }
 
 impl Drop for AllUsers {
     fn drop(&mut self) {
+        #[cfg(not(target_os = "android"))]
         unsafe { endpwent() };
     }
 }
@@ -428,6 +434,11 @@ impl Drop for AllUsers {
 impl Iterator for AllUsers {
     type Item = User;
 
+    #[cfg(target_os = "android")]
+    fn next(&mut self) -> Option<User> {
+        None
+    }
+    #[cfg(not(target_os = "android"))]
     fn next(&mut self) -> Option<User> {
         unsafe { passwd_to_user(getpwent()) }
     }
@@ -457,7 +468,7 @@ pub mod os {
     /// Although the `passwd` struct is common among Unix systems, its actual
     /// format can vary. See the definitions in the `base` module to check which
     /// fields are actually present.
-    #[cfg(any(target_os = "linux", target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd", target_os = "solaris"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd", target_os = "solaris"))]
     pub mod unix {
         use std::ffi::{OsStr, OsString};
         use std::path::{Path, PathBuf};
@@ -531,18 +542,25 @@ pub mod os {
             /// Extract the OS-specific fields from the C `passwd` struct that
             /// we just read.
             pub unsafe fn from_passwd(passwd: c_passwd) -> Self {
-                let home_dir = from_raw_buf(passwd.pw_dir).into();
-                let shell    = from_raw_buf(passwd.pw_shell).into();
-                let password = from_raw_buf(passwd.pw_passwd);
+                #[cfg(target_os = "android")]
+                {
+                    Default::default()
+                }
+                #[cfg(not(target_os = "android"))]
+                {
+                    let home_dir = from_raw_buf(passwd.pw_dir).into();
+                    let shell    = from_raw_buf(passwd.pw_shell).into();
+                    let password = from_raw_buf(passwd.pw_passwd);
 
-                UserExtras { home_dir, shell, password }
+                    UserExtras { home_dir, shell, password }
+                }
             }
         }
 
-        #[cfg(any(target_os = "linux", target_os = "solaris"))]
+        #[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
         use super::super::User;
 
-        #[cfg(any(target_os = "linux", target_os = "solaris"))]
+        #[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
         impl UserExt for User {
             fn home_dir(&self) -> &Path {
                 Path::new(&self.extras.home_dir)
@@ -703,11 +721,11 @@ pub mod os {
     pub type UserExtras = bsd::UserExtras;
 
     /// Any extra fields on a `User` specific to the current platform.
-    #[cfg(any(target_os = "linux", target_os = "solaris"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
     pub type UserExtras = unix::UserExtras;
 
     /// Any extra fields on a `Group` specific to the current platform.
-    #[cfg(any(target_os = "linux", target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd", target_os = "solaris"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "macos", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd", target_os = "solaris"))]
     pub type GroupExtras = unix::GroupExtras;
 }
 

